scripts/ci/pr-bot/shared/userCommand.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const github = require("./githubUtils");
+const commentStrings = require("./commentStrings");
+const { BOT_NAME } = require("./constants");
+const { StateClient } = require("./persistentState");
+const { ReviewerConfig } = require("./reviewerConfig");
+
+// Reads the comment and processes the command if one is contained in it.
+// Returns true if it runs a command, false otherwise.
+export async function processCommand(
+  payload: any,
+  commentAuthor: string,
+  commentText: string,
+  stateClient: typeof StateClient,
+  reviewerConfig: typeof ReviewerConfig
+) {
+  // Don't process any commands from our bot.
+  if (commentAuthor === BOT_NAME) {
+    return false;
+  }
+  console.log(commentAuthor);
+
+  const pullNumber = payload.issue?.number || payload.pull_request?.number;
+  commentText = commentText.toLowerCase();
+  if (commentText.indexOf("r: @") > -1) {
+    await manuallyAssignedToReviewer(pullNumber, stateClient);
+  } else if (commentText.indexOf("assign to next reviewer") > -1) {
+    await assignToNextReviewer(
+      payload,
+      commentAuthor,
+      pullNumber,
+      stateClient,
+      reviewerConfig
+    );
+  } else if (commentText.indexOf("stop reviewer notifications") > -1) {
+    await stopReviewerNotifications(
+      pullNumber,
+      stateClient,
+      "requested by reviewer"
+    );
+  } else if (commentText.indexOf("remind me after tests pass") > -1) {
+    await remindAfterTestsPass(pullNumber, commentAuthor, stateClient);
+  } else if (commentText.indexOf("waiting on author") > -1) {
+    await waitOnAuthor(payload, pullNumber, stateClient);
+  } else if (commentText.indexOf("assign set of reviewers") > -1) {
+    await assignReviewerSet(payload, pullNumber, stateClient, reviewerConfig);
+  } else {
+    return false;
+  }
+
+  return true;
+}
+
+async function assignToNextReviewer(
+  payload: any,
```

Comment:
`any` => `unknown`

Reply:
I skipped this one because we don't have the ability to intelligently cast this later on. This really is an `any` value because it represents a payload with potentially different forms - its much easier to pass it around like that than continually try to cast it to different things everywhere we want to dereference a value (which basically loses any type safety anyways)